plugins/telemetry/telemetry.h

```diff
@@ -87,100 +141,458 @@ class Telemetry
         UNKNOWN
     };
 
+    /**
+     * Returns human-readable English string for Telemetry::Result.
+     *
+     * @param result result enum
+     */
     static const char *result_str(Result result);
 
+    /**
+     * Callback type for telemetry requests.
+     */
     typedef std::function<void(Result)> result_callback_t;
 
+    /**
```

Comment:
For all of these, are there any limits/guidance?

Reply:
This depends on the link bandwidth, so it's a bit more complicated I'm afraid.